checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakVisitor.java

```diff
@@ -0,0 +1,226 @@
+package org.checkerframework.checker.resourceleak;
+
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.VariableTree;
+import java.util.ArrayList;
+import java.util.List;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import org.checkerframework.checker.calledmethods.CalledMethodsVisitor;
+import org.checkerframework.checker.calledmethods.qual.EnsuresCalledMethods;
+import org.checkerframework.checker.mustcall.CreatesMustCallForElementSupplier;
+import org.checkerframework.checker.mustcall.MustCallAnnotatedTypeFactory;
+import org.checkerframework.checker.mustcall.MustCallChecker;
+import org.checkerframework.checker.mustcall.qual.CreatesMustCallFor;
+import org.checkerframework.checker.mustcall.qual.Owning;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.framework.type.GenericAnnotatedTypeFactory;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.ElementUtils;
+import org.checkerframework.javacutil.TreeUtils;
+
+/**
+ * The visitor for the Resource Leak Checker. Responsible for some {@link CreatesMustCallFor}
+ * checking.
+ */
+public class ResourceLeakVisitor extends CalledMethodsVisitor {
+
+  /**
+   * Because CalledMethodsVisitor doesn't have a type parameter, we need a reference to the type
+   * factory that has this static type to access the features that ResourceLeakAnnotatedTypeFactory
+   * implements but CalledMethodsAnnotatedTypeFactory does not.
+   */
+  private final ResourceLeakAnnotatedTypeFactory rlTypeFactory;
+
+  /**
+   * Create the visitor.
+   *
+   * @param checker the type-checker associated with this visitor
+   */
+  public ResourceLeakVisitor(final BaseTypeChecker checker) {
+    super(checker);
+    rlTypeFactory = (ResourceLeakAnnotatedTypeFactory) atypeFactory;
+  }
+
+  @Override
+  protected ResourceLeakAnnotatedTypeFactory createTypeFactory() {
+    return new ResourceLeakAnnotatedTypeFactory(checker);
+  }
+
+  @Override
+  public Void visitMethod(MethodTree node, Void p) {
+    ExecutableElement elt = TreeUtils.elementFromDeclaration(node);
+    MustCallAnnotatedTypeFactory mcAtf =
```

Comment:
Store this in a field.  That will reduce the need to compute it and to pass it around.

Reply:
No. The documentation of `getTypeFactoryOfSubchecker` says: "Because the visitor state is copied, call this method each time a subfactory is needed rather than store the returned subfactory in a field."